test/file.js

```diff
@@ -2799,15 +2956,44 @@ describe('File', function() {
   });
 
   describe('rotateEncryptionKey', function() {
+    it('should create new File correctly', function(done) {
+      var options = {};
+
+      file.bucket.file = function(id, options_) {
+        assert.strictEqual(id, file.id);
+        assert.strictEqual(options_, options);
+        done();
+      };
+
+      file.rotateEncryptionKey(options, assert.ifError);
+    });
+
+    it('should default to customer-supplied encryption key', function(done) {
+      var encryptionKey = 'encryption-key';
+
+      file.bucket.file = function(id, options) {
+        assert.strictEqual(options.encryptionKey, encryptionKey);
+        done();
+      };
+
+      file.rotateEncryptionKey(encryptionKey, assert.ifError);
```

Comment:
If the design is kept for rotateEncryptionKey(), there should exist a test that shows how to use a CSEK with a `{}` along with a string option.

Reply:
That's actually covered in this test:

```js
it('should create new File correctly', function(done) {
  var options = {};

  file.bucket.file = function(id, options_) {
    assert.strictEqual(id, file.id);
    assert.strictEqual(options_, options);
    done();
  };

  file.rotateEncryptionKey(options, assert.ifError);
});
```

That tests that *any* object goes through, which includes one with `encryptionKey` or any arbitrary property name.

Then we have "should default to customer-supplied encryption key" for strings, and "should accept a Buffer for customer-supplied encryption key" for Buffers

Reply:
Just for context, here's the full source code of the method we're testing:

```js
File.prototype.rotateEncryptionKey = function(options, callback) {
  if (!is.object(options)) {
    options = {
      encryptionKey: options,
    };
  }

  var newFile = this.bucket.file(this.id, options);
  this.copy(newFile, callback);
};
```

Since there's no specific handling when `options` is an object, we don't have any special tests other than that the original user-provided object makes its way to the `this.bucket.file` constructor.

We have system tests that cover that rotating works with CSEK and KMS.

Reply:
We don't explicitly specify `{ encryptionKey: '...' }` in a unit test, because there is no line of source code that would care if we put "encryptionKey" or "kmsKeyName" or "madeUpKey", etc. Our testing strategy goes line-by-line to test that the user input goes to the right place.

In this case, the source code takes the user-provided options object and passes it directly to `this.bucket.file(this.id, options)`. So our test confirms that `this.bucket.file` receives the exact user-provided object. It's the the job of `this.bucket.file` to confirm the right thing happens next, which we have a separate collection of tests for.

To test that first block of code in the function:

```js
if (!is.object(options)) {
  options = {
    encryptionKey: options,
  };
}
```

We have these two tests:

```js
it('should default to customer-supplied encryption key', function(done) {
  var encryptionKey = 'encryption-key';

  file.bucket.file = function(id, options) {
    assert.strictEqual(options.encryptionKey, encryptionKey);
    done();
  };

  file.rotateEncryptionKey(encryptionKey, assert.ifError);
});

it('should accept a Buffer for customer-supplied encryption key', function(done) {
  var encryptionKey = crypto.randomBytes(32);

  file.bucket.file = function(id, options) {
    assert.strictEqual(options.encryptionKey, encryptionKey);
    done();
  };

  file.rotateEncryptionKey(encryptionKey, assert.ifError);
});
```

Those confirm that what we list in our JSDocs params work as advertised. The first gives it a string, then confirms it is interpreted as an `encryptionKey`. The second gives a buffer, and confirms the same.

Reply:
Ah okay, so the test verifies that any `{}` is received correctly in a request. I thought there was a test for `{ encryptionKey: '...' }` that I wasn't seeing. Thanks!